src/types.py

```diff
@@ -24,77 +25,25 @@
 AccessPath = List[Union[str, int]]
 
 
-@dataclass(eq=False)
-class TypeUniverse:
+class GlobalTypeInfo:
     """
-    Mutable shared state for Types, currently maintaining the set of available
-    struct types.
-    Unlike TypeMap, which is immutable and can be reused across tests/threads,
-    a new TypeUniverse instance should be created for each isolated run.
-    However, if the object is reused between functions, it may help provide
-    cross-function type resolution.
+    Mutable shared state for Types, implemented by GlobalInfo in translate.py.
```

Comment:
Hrm, this circular reference to translate.py actually rather bothers me -- it makes types.py no longer self contained. I wonder if the right split may actually be info about structs in types.py, and info about variables in translate.py.

Reply:
Refactored again in `2b420a6` -- I think it's converging on a good solution? :smile: